nemo/src/builder_proxy.rs

```diff
@@ -11,286 +18,549 @@ use nemo_physical::{
-        <LogicalAnyColumnBuilderProxy<'_, '_> as ColumnBuilderProxy<TurtleEncodedRDFTerm>>::commit(
-            self,
-        );
+    fn add(&mut self, input: i64) -> Result<(), ReadingError> {
+        <Self as ColumnBuilderProxy<i64>>::commit(self);
+        self.inner.add(logical_integer_to_physical_string(input))
+    }
+}
+
+impl ColumnBuilderProxy<Double> for LogicalAnyColumnBuilderProxy<'_, '_> {
+    logical_generic_trait_impl!();
 
-        self.physical.add(input.into_normalized_string())
+    fn add(&mut self, input: Double) -> Result<(), ReadingError> {
+        <Self as ColumnBuilderProxy<Double>>::commit(self);
+        self.inner.add(logical_double_to_physical_string(input))
     }
 }
 
-impl<'a, 'b> LogicalColumnBuilderProxy<'a, 'b> for LogicalAnyColumnBuilderProxy<'a, 'b> {
-    fn new(physical_builder_proxy: &'b mut PhysicalBuilderProxyEnum<'a>) -> Self {
+/// [`LogicalColumnBuilderProxy`] to add String
+#[derive(Debug)]
+pub struct LogicalStringColumnBuilderProxy<'a: 'b, 'b> {
+    inner: &'b mut PhysicalStringColumnBuilderProxy<'a>,
+}
+
+impl<'a, 'b> LogicalStringColumnBuilderProxy<'a, 'b> {
+    pub fn new(physical_builder_proxy: &'b mut PhysicalBuilderProxyEnum<'a>) -> Self {
         match physical_builder_proxy {
-            PhysicalBuilderProxyEnum::String(physical) => Self { physical },
+            PhysicalBuilderProxyEnum::String(inner) => Self { inner },
             _ => unreachable!("If the database representation of the logical types is correct, we never reach this branch.")
         }
     }
+
+    pub fn into_parser<Intermediate>(self) -> GenericLogicalParser<Intermediate, Self>
+    where
+        Self: ColumnBuilderProxy<Intermediate>,
+    {
+        GenericLogicalParser::new(self)
+    }
 }
 
-/// [`LogicalColumnBuilderProxy`] to add String
-#[derive(Debug)]
-pub struct LogicalStringColumnBuilderProxy<'a: 'b, 'b> {
-    physical: &'b mut PhysicalStringColumnBuilderProxy<'a>,
+impl ColumnBuilderProxy<Term> for LogicalStringColumnBuilderProxy<'_, '_> {
+    logical_generic_trait_impl!();
+
+    fn add(&mut self, input: Term) -> Result<(), ReadingError> {
+        <Self as ColumnBuilderProxy<Term>>::commit(self);
+        self.inner.add(any_string_to_physical_string(input)?)
+    }
 }
 
 impl ColumnBuilderProxy<String> for LogicalStringColumnBuilderProxy<'_, '_> {
     logical_generic_trait_impl!();
 
     fn add(&mut self, input: String) -> Result<(), ReadingError> {
-        self.commit();
+        <Self as ColumnBuilderProxy<String>>::commit(self);
         // NOTE: we just pipe the string through as is, in particular we do not parse potential RDF terms
         // NOTE: we store the string in the same format as it would be stored in an any column;
         // this is important since right now we sometimes use the LogicalStringColumnBuilderProxy to directly write data that is known to only be strings into an any column and not only into string columns
```

Comment:
This is no longer true now, is it?

Reply:
I think the first two lines are still kind of true but the comment does not really fit here anymore so I will remove it.